test-helpers/src/connection/kafka/cpp.rs

```diff
@@ -157,6 +158,68 @@ impl KafkaConsumerCpp {
             offset: Some(message.offset()),
         }
     }
+
+    /// The offset to be committed should be lastProcessedMessageOffset + 1.
+    pub fn commit(
+        &self,
+        offsets: &HashMap<TopicPartition, i64>,
+        commit_mode: rdkafka::consumer::CommitMode,
+    ) {
+        let offsets_map: HashMap<(String, i32), rdkafka::Offset> = offsets
+            .iter()
+            .map(|(tp, offset)| {
+                (
+                    (tp.topic_name.clone(), tp.partition),
+                    rdkafka::Offset::Offset(*offset),
+                )
+            })
+            .collect();
+
+        let offsets_list = rdkafka::TopicPartitionList::from_topic_map(&offsets_map).unwrap();
+
+        self.consumer.commit(&offsets_list, commit_mode).unwrap();
+    }
+
+    pub fn committed_offsets(
+        &self,
+        partitions: HashSet<TopicPartition>,
+    ) -> HashMap<TopicPartition, i64> {
+        let mut offsets = HashMap::new();
+        let mut tpl = rdkafka::TopicPartitionList::with_capacity(partitions.len());
+
+        // This TopicPartitionList is used to query the committed offsets for the partitions
+        // Hence offset is set to Invalid
+        for tp in &partitions {
+            tpl.add_partition_offset(
+                tp.topic_name.as_str(),
+                tp.partition,
+                rdkafka::Offset::Invalid,
+            )
+            .expect("Failed to add the topic and partition");
+        }
+
+        let committed_offsets = self
+            .consumer
+            .committed_offsets(tpl, Timeout::After(Duration::from_secs(30)))
+            .unwrap();
+
+        for tp in &partitions {
```

Comment:
I think this can be simplified:

![image](https://github.com/user-attachments/assets/1c6319f9-0e2d-42a5-8a33-7aceab103b93)